apisix/core/config_yaml.lua

```diff
@@ -274,7 +275,7 @@ local function sync_data(self)
                           ", it should be an object")
             end
 
-            local id = item.id or ("arr_" .. idx)
+            local id = item.id or item.username or ("arr_" .. idx)
```

Comment:
Is this change necessary? We have already set `id` to `username` for consumer in `apisix/consumer.lua`.
```
consumer.value.id = consumer.value.username
```

Reply:
If not, the default is values_hash ["arr_1"] = consumer{}, 

The filter is called after the cache is set. In the following calls, it will not get the consumer


<img width="536" alt="image" src="https://github.com/user-attachments/assets/92bccbc0-3988-46e1-91cc-6865a423b238" />


<img width="391" alt="image" src="https://github.com/user-attachments/assets/767d8923-8db2-43a0-9378-60f602719cab" />